### ISSUE TITLE
Redudant @var/@type/@param

### ISSUE THREAD

**Issue by rentalhost** [kalessil/phpinspectionsea#1443]
Description (screenshot):
--
PS supports "redundant suppression". I think a good idea supports "redundant @var" and similar. For instance, if a method always returns `User|null`, so `@var User|null` is redundant.

Example
--

```php
function getUser(): ?User { ... }

/** @var User|null $user redundant! */
$user = getUser();

/** @var User $user allowed */
$user = getUser();
```

**Comment by funivan** [comment]
I wrote simple plugin which contains this feature https://github.com/funivan/phpclean#redundantdoccommenttag
You can test it already.

It can be simply implemented in this repository if needed.

**Comment by orklah** [comment]
> I wrote simple plugin which contains this feature https://github.com/funivan/phpclean#redundantdoccommenttag
> You can test it already.
> 
> It can be simply implemented in this repository if needed.

Would it be possible to implement the same thing but for ```@var``` related to variables in code? IIRC, your inspection flag only the doc blocks related to functions, methods or attributes.

**Comment by funivan** [comment]
@orklah So you write var for the variable and want to remove it?

**Comment by orklah** [comment]
Yeah. It happens a lot, you write a @var because the IDE doesn't resolve a type, but later you add a type somewhere else(in a return hint for example) and the @var become useless.

**Comment by kalessil** [comment]
@funivan, check your link - looks pretty much the same pattern. Cna you incordporate this in your plugin, please?

**Comment by kalessil** [comment]
If yes, feel free to close this ticket.

**Comment by funivan** [comment]
Sorry for the delay.
I have created an issue in the PhpClean repository.

**Comment by kalessil** [comment]
No worries, thank you!

**Comment by orklah** [comment]
FYI, Issue was implemented in PhpStorm 2020.1
